daemons/invoke_lambda.sh

```diff
@@ -2,16 +2,29 @@
 
 set -euo pipefail
 
-if [[ $# != 3 ]]; then
-    echo "Usage: $(basename $0) daemon-name stage lambda-input-file"
+if [[ $# != 4 ]]; then
+    echo "Usage: $(basename $0) daemon-name stage lambda-input-file bundle-file"
     exit 1
 fi
 
 lambda_name="$1-$2"
 lambda_input_file=$3
+bundle_file="$4"
 
-lambda_payload="$(sed "s/testBucket/$DSS_S3_TEST_BUCKET/g" $lambda_input_file)"
-lambda_output="$(aws lambda invoke --function-name $lambda_name --invocation-type RequestResponse --payload "$lambda_payload" --log-type Tail /dev/stdout | sed 's/^null//')"
+BUNDLE_KEY="bundles/$(basename "${bundle_file}")"
+
+aws s3 cp "${bundle_file}" s3://${DSS_S3_TEST_BUCKET}/"${BUNDLE_KEY}"
+BUNDLE_FILE_ETAG=$(aws s3api head-object --bucket ${DSS_S3_TEST_BUCKET} --key "${BUNDLE_KEY}" | jq -r '.ETag | fromjson')
+BUNDLE_FILE_SIZE=$(cat "${bundle_file}" | wc -c)
+
+# the wonky if-else is required because us-east-1 is represented as a null location constraint.  weird, eh?
+DSS_S3_BUCKET_REGION=$(aws s3api get-bucket-location --bucket ${DSS_S3_TEST_BUCKET} | jq -r 'if (.LocationConstraint == null) then "us-east-1" else .LocationConstraint end')
```

Comment:
What are the circumstances in which a bucket is expected not to be in the same region as our deployment? This seems like overparameterizing the setup. If this bucket is in a remote region, the app is misconfigured, no?

Consider replacing this with simply `aws configure get region`.